### DISCUSSION TITLE
[Feature] Allow alternative download URLs

### DISCUSSION THREAD

**Discussion by NSP-0123456** [ScoopInstaller/Scoop#5017] [Ideas]
I have some application where i can install some flavors 32bit, 64bit, 32+64bit, compatibleMode i can define all in my manifest. Currently all architecture/flavor name are valid in json manifest for applications. (I can use checkvers and autoupdate....)

But when i want to install, i'm limited to only two values 32bit/64bit could it be possible to accept with -a additional verb and just print a warning instead of quitting ?

in core.ps1 in function ensure_architecture removing default behavior to return $architecture_opt seems to be ok for next step ! 
Anyhow, if application do not support exotic flavor installation is not made !

**Comment by rashil2000** [comment]
> 32+64bit

Isn't this just 32bit?

> compatibleMode

Which type of architecture is this?


Regardless, only those architectures will be supported which can be determined from the OS. Neither of these can be.

**Comment by NSP-0123456** [comment]
32+64 is more to install both 32 and 64bit pgm in same folder for a 64bit cpu on 64bit os.
compatibleMode is just a verb do not stick to cpu architecture just consider it as different flavor of a single application i want to keep in a single json and deployed name some pgm have different install file for different languages  i waouls have used 32bit_en, 64bit_fr, 64bit_tk 32bit_sp, 64bit_de ...

**Comment by rashil2000** [comment]
> 32+64 is more to install both 32 and 64bit pgm in same folder for a 64bit cpu on 64bit os.

See https://github.com/ScoopInstaller/Scoop/issues/3265

> compatibleMode is just a verb do not stick to cpu architecture just consider it as different flavor of a single application i want to keep in a single json and deployed name

Can you give an example?

> some pgm have different install file for different languages i waouls have used 32bit_en, 64bit_fr, 64bit_tk 32bit_sp, 64bit_de ...

For this you need to create separate manifests as languages are not architectures.

**Reply by NSP-0123456** [reply]
I know that i can write as many json manifest as i need, and this is not what i want  !

The goal of the feature request is to not have not a architecture but a "verb" to have multiple flavor of a single application in a single json.  

If you are picky about architecture to have to be an architecture then add a new part in the json to be able to select another thing than architecture !
